backend/routes/XrplNFTHelper.js

```diff
@@ -0,0 +1,462 @@
+console.log("Error occured during batchX() call" + err)
+console.log(err.data)
+return;
+}
+
+
+}
+
+
+
+  /*getTokens
+  *
+  *@returns array of NFTokenID's
+  */
+  async getTokensFromLedger(){
+    try {
+  
+      
+        const client = new xrpl.Client(this.clientDetails)
+        await client.connect()
+
+        console.log("Connected to Sandbox..getting all NFT's.****")
+  
+      
+        let nfts = await client.request({
+          method: "account_nfts",
+          account: this.transactionDetails.Account
+          })
+  
+  
+          await client.disconnect()
+          console.log("disconnecting")
+  
+          return nfts.result.account_nfts
+  
+  
+  }
+  catch(err) {
+  console.log("Error occured during getTokens() call" + err)
+  return;
+  }
+  
+  }
+
+
+/* Burn specified NFT
+* Params (required): 
+          - TransactionType: this.transactionDetails.TransactionType,
+          - Account: this.transactionDetails.Account,
+          - NFTokenID: this.transactionDetails.NFTokenID
+
+  Returns: Transaction Result string.
+*
+*/
+  async burnNFT(){
+    try {
+     
+
+      const wallet = xrpl.Wallet.fromSeed(this.transactionDetails.Secret)
+      const client = new xrpl.Client(this.clientDetails)
+      await client.connect()
+
+      console.log("Connected to Sandbox..burning single NFT.")
+
+   
+    
+    const transactionData = {
+          TransactionType: this.transactionDetails.TransactionType,
+          Account: this.transactionDetails.Account,
+          NFTokenID: this.transactionDetails.NFTokenID
+      }
+      
+    
+      
+      const tx = await client.submitAndWait(transactionData,{wallet})
+  
+      client.disconnect()
+      return tx.result.meta.TransactionResult
+
+}
+catch(err) {
+  console.log("Error occured during minToken() call" + err)
+  return;
+}
+  
+  }
+
+
+  /*Burn all NFTs in the account
+
+  Params (required): 
+  - transactionDetails.Secret
+  - transactionDetails.Account
+
+  Returns: 
+  Array of NFTokenID's for removal of metadata storage
+
+
+  */
+
+
+  async burnAllNFT(){
+    try {
+     
+      const wallet = xrpl.Wallet.fromSeed(this.transactionDetails.Secret)
+      const client = new xrpl.Client(this.clientDetails)
+      await client.connect()
+
+      console.log("Connected to Sandbox..burning ALL NFT's for specified account.")
+
+      let nfts = await client.request({
+        method: "account_nfts",
+        account: this.transactionDetails.Account
+        })
+
+        console.log("Attempting to burn " + nfts.result.account_nfts.length + " NFT's..")
+              
+        for (let index = 0; index < nfts.result.account_nfts.length; index++) {
+
+
+          const transactionData = {
+            TransactionType: this.transactionDetails.TransactionType,
+            Account: this.transactionDetails.Account,
+            NFTokenID: nfts.result.account_nfts[index].NFTokenID
+        }
+
+
+          const tx = await client.submitAndWait(transactionData,{wallet})

```

Comment:
What should happen if there's an error in burning one of the transactions? (Should it keep going, or should it stop?)